### ISSUE TITLE
heat map of safe and high-risk locations

### ISSUE THREAD

**Issue by andreiyurkevich** [enigmampc/SafeTrace#9]
There is a suggestion for the product roadmap. It would be useful to track the history of places visited by people who were later diagnosed on the disease and create a heat map. The system will have a "score" of places based on the number of visits by probably contagious people. The application can provide information about "clean" areas and the areas where additional caution is recommended.

**Comment by cankisagun** [comment]
History of places visited by all users are kept in the DB. 
- If you haven't posted positively, we need your location history so that if someone who's reported positive after you, have been in close proximity with you, you can get a notification.
- You have the option to log back in and change your status (positive / negative), while your location history is being stored (for 14 days - current working assumption)
- If you have tested positive your history is stored, you can always add the most recent location history as a user

**Comment by ainsleys** [comment]
one of the challenges of marking an area "clean" is that it risks giving users a false sense of security, which we must avoid. But agree with you completely on the value of a heatmap!

**Comment by tuneerguha** [comment]
1. @cankisagun As for the history of places, we could use a framework like OpenDMTP(We could possibly modify a software like openGTS to suit the req). it is primarily meant for vehicular fleet management, but that works as an advantage since it would let people track positive/negative cases. It also allows geo-fencing so people moving in and out of zones can be notified. 

2. @ainsleys -  We could use visual stimuli to show that a clean area may not actually be clean. Usually heat-maps have hot and cold zones shown in a spectrum of blue and red, we could use colors that don't give a complete sense of security, like a spectrum of light orange to bright red.

**Comment by cankisagun** [comment]
@tuneerguha I feel like the applications that are interacting with our DB needs to implement openGTS as the web portal we are building depends on Google Maps for location data. Am I missing something?

**Comment by tuneerguha** [comment]
@cankisagun Not really, It has support for OpenStreetMap. It may not be as accurate as google maps, but i guess its good enough.
